src/devices/cpu/i386/i386.cpp

```diff
@@ -2025,7 +2025,6 @@ void i386_device::i386_common_init()
 	m_ferr_handler(0);
 
 	set_icountptr(m_cycles);
-	m_notifier = m_program->add_change_notifier([this] (read_or_write mode) { dri_changed(); });
```

Comment:
If you remove this, the CPU core won’t be notified if handlers are changed in the address space, possibly requiring passthrough handlers to be reinstalled.

Reply:
@cuavas Firstly, sorry for causing such a mess of this pull request :(
OK, having refreshed my memory the problem with the notifier is that it seems to causes dri_changed() to be called while actually within dri_changed() :(
What happenS is the adding of the read/write tap within dri_changed causes dri_changed to be called which can't be right. Placing debug in dri_changed at the top and bottom to show "enter" and "exit" produces the following output when the taps are changed...
enter, enter, exit, exit.
Further debug shows the point of the second entry as when the tap is installed.
How to prevent this if dri_changed really needs to be called when handlers are changed ?

Reply:
You have to guard against installing taps recursively.  For example see the watchpoint code in https://github.com/mamedev/mame/blob/master/src/emu/debug/points.cpp